GUI/MainModList.cs

```diff
@@ -521,21 +521,28 @@ public IEnumerable<DataGridViewRow> ConstructModList(IEnumerable<GUIMod> modules
                 full_list_of_mod_rows.Remove(mod.Identifier);
                 var item = new DataGridViewRow {Tag = mod};
 
-                var selecting = mod.IsInstallable()
-                    ? (DataGridViewCell) new DataGridViewCheckBoxCell()
-                    : new DataGridViewTextBoxCell();
-
-                selecting.Value = mod.IsInstallable()
-                    ? (object) mod.IsInstalled
-                    : (mod.IsAutodetected ? "AD" : "-");
-
-                var updating = mod.HasUpdate && !mod.IsAutodetected
-                    ? new DataGridViewCheckBoxCell()
-                    : (DataGridViewCell) new DataGridViewTextBoxCell();
+                ModChange myChange = mc?.Find((ModChange ch) => ch.Mod.Identifier == mod.Identifier);
 
-                updating.Value = !mod.IsInstallable() || !mod.HasUpdate
-                    ? "-"
-                    : (object) false;
+                var selecting = mod.IsInstallable()
+                    ? (DataGridViewCell) new DataGridViewCheckBoxCell() {
+                        Value = myChange == null ? mod.IsInstalled
+                            : myChange.ChangeType == GUIModChangeType.Install ? true
+                            : myChange.ChangeType == GUIModChangeType.Remove  ? false
+                            : mod.IsInstalled
+                    } : new DataGridViewTextBoxCell() {
+                        Value    = mod.IsAutodetected ? "AD" : "-",
+                        ReadOnly = true
```

Comment:
This line causes an exception (Windows):

Unhandled exception:
System.InvalidOperationException: ReadOnly property of a cell cannot be set before it is added to a row.
   at System.Windows.Forms.DataGridViewCell.set_ReadOnly(Boolean value)
   at CKAN.MainModList.ConstructModList(IEnumerable`1 modules, List`1 mc, Boolean refreshAll, Boolean hideEpochs) in S:\KSP\CKAN\CKAN\GUI\MainModList.cs:line 537
   at CKAN.Main._UpdateModsList(Boolean repo_updated, List`1 mc) in S:\KSP\CKAN\CKAN\GUI\MainModList.cs:line 186
   at CKAN.Main.<>c__DisplayClass233_0.<UpdateModsList>b__0() in S:\KSP\CKAN\CKAN\GUI\MainModList.cs:line 143
   at CKAN.Util.Invoke[T](T obj, Action action) in S:\KSP\CKAN\CKAN\GUI\Util.cs:line 28
   at CKAN.Main.UpdateModsList(Boolean repo_updated, List`1 mc) in S:\KSP\CKAN\CKAN\GUI\MainModList.cs:line 143
   at CKAN.Main.CurrentInstanceUpdated() in S:\KSP\CKAN\CKAN\GUI\Main.cs:line 441
   at CKAN.Main.OnLoad(EventArgs e) in S:\KSP\CKAN\CKAN\GUI\Main.cs:line 347
   at System.Windows.Forms.Form.OnCreateControl()
   at System.Windows.Forms.Control.CreateControl(Boolean fIgnoreVisible)
   at System.Windows.Forms.Control.CreateControl()
   at System.Windows.Forms.Control.WmShowWindow(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Form.WmShowWindow(Message& m)
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)